core/node/da_clients/src/avail/client.rs

```diff
@@ -35,37 +108,165 @@ impl DataAvailabilityClient for AvailClient {
+            .as_ref()
+            .unwrap()
             .build_extrinsic(&client, data)
             .await
             .map_err(to_non_retriable_da_error)?;
 
         let block_hash = self
             .sdk_client
+            .as_ref()
+            .unwrap()
             .submit_extrinsic(&client, extrinsic.as_str())
             .await
             .map_err(to_non_retriable_da_error)?;
         let tx_id = self
             .sdk_client
+            .as_ref()
+            .unwrap()
             .get_tx_id(&client, block_hash.as_str(), extrinsic.as_str())
             .await
             .map_err(to_non_retriable_da_error)?;
-
         Ok(DispatchResponse::from(format!("{}:{}", block_hash, tx_id)))
     }
 
     async fn get_inclusion_data(
         &self,
-        _blob_id: &str,
+        blob_id: &str,
     ) -> anyhow::Result<Option<InclusionData>, DAError> {
-        // TODO: implement inclusion data retrieval
-        Ok(Some(InclusionData { data: vec![] }))
+        let (block_hash, tx_idx) = blob_id.split_once(':').ok_or_else(|| DAError {
+            error: anyhow!("Invalid blob ID format"),
+            is_retriable: false,
+        })?;
+
+        let url = format!(
+            "{}/eth/proof/{}?index={}",
+            self.config.bridge_api_url, block_hash, tx_idx
+        );
+        let mut response: reqwest::Response;
+        let mut retries = self.config.max_retries;
+        let mut response_text: String;
+        let mut bridge_api_data: BridgeAPIResponse;
+        loop {
+            response = self
+                .api_client
+                .get(&url)
+                .send()
+                .await
+                .map_err(to_retriable_da_error)?;
+            response_text = response.text().await.unwrap();
+
+            if let Ok(data) = serde_json::from_str::<BridgeAPIResponse>(&response_text) {
+                bridge_api_data = data;
+                if bridge_api_data.error.is_none() {
+                    break;
+                }
+            }
+
+            tokio::time::sleep(tokio::time::Duration::from_secs(
+                u64::try_from(480).unwrap(),
+            ))
+            .await; // usually takes 15 mins on Hex
+            retries += 1;
+            if retries > self.config.max_retries {
+                return Err(DAError {
+                    error: anyhow!("Failed to get inclusion data"),
+                    is_retriable: true,
+                });
+            }
+        }
+        let attestation_data: MerkleProofInput = MerkleProofInput {
+            dataRootProof: bridge_api_data.data_root_proof.unwrap(),
+            leafProof: bridge_api_data.leaf_proof.unwrap(),
+            rangeHash: bridge_api_data.range_hash.unwrap(),
+            dataRootIndex: bridge_api_data.data_root_index.unwrap(),
+            blobRoot: bridge_api_data.blob_root.unwrap(),
+            bridgeRoot: bridge_api_data.bridge_root.unwrap(),
+            leaf: bridge_api_data.leaf.unwrap(),
```

Comment:
is this a simple `keccak(data)` same as in L2DAValidator contract?

Reply:
That is correct.